pkg/vpcmodel/abstractVPC.go

```diff
@@ -249,20 +249,26 @@ type RulesInTable struct {
 
 // RuleOfFilter a single rule in filter given the layer (SGLayer/NACLLayer)
 type RuleOfFilter struct {
-	LayerName   string             `json:"layer"`
-	FilterName  string             `json:"table"` // todo: use map from FilterIndex to FilterName and delete?
-	FilterIndex int                `json:"-"`
-	RuleIndex   int                `json:"rule_index"`
-	RuleDesc    string             `json:"rule_description"`
-	IPBlocks    []*ipblock.IPBlock `json:"ip_blocks,omitempty"` // CIDR/IPBlocks referenced by the rule (src/dst/local...)
+	Filter    Filter
+	RuleIndex int                `json:"rule_index"`
+	RuleDesc  string             `json:"rule_description"`
+	IPBlocks  []*ipblock.IPBlock `json:"ip_blocks,omitempty"` // CIDR/IPBlocks referenced by the rule (src/dst/local...)
+}
+
+type Filter struct {
+	LayerName   string `json:"layer"`
+	FilterName  string `json:"table"`
+	FilterIndex int    `json:"-"`
 }
 
 func NewRuleOfFilter(layerName, filterName, desc string, filterIndex, ruleIndex int, ipBlocks []*ipblock.IPBlock) *RuleOfFilter {
-	return &RuleOfFilter{LayerName: layerName, FilterIndex: filterIndex, FilterName: filterName,
-		RuleIndex: ruleIndex, RuleDesc: desc,
-		IPBlocks: ipBlocks}
+	table := Filter{LayerName: layerName, FilterIndex: filterIndex, FilterName: filterName}
+	return &RuleOfFilter{Filter: table, RuleIndex: ruleIndex, RuleDesc: desc, IPBlocks: ipBlocks}
 }
 
+// FiltersToAttached map from each filter to the resources attached to it; e.g. from NACL to subnets
+type FiltersToAttached map[Filter][]VPCResourceIntf
```

Comment:
rename `FiltersToAttached` to `FiltersToAttachedResources` or `FiltersAttachedResources`